### ISSUE TITLE
HepMC plugin looks incomplete

### ISSUE THREAD

**Issue by andiwand** [acts-project/acts#3050]
While working on https://github.com/acts-project/acts/pull/2998 I noticed that HepMC plugin is incomplete. It also seems like we are not using or testing the code anywhere in detail.

- Problem with ownership of Acts particles
  - Currently Acts particles cannot refer to each other and are usually owned by a container holding all particles
  - When introducing vertices this becomes a problem
  - When modeling vertices this becomes a problem
- Conversion is incomplete
  - The Acts barcode is not properly filled
- Code is not used
- Code is not properly tested

**Comment by AJPfleger** [comment]
@paulgessinger:
The plugin is quite old and was used by @FabianKlimpel in 2018 for testing.
Maybe there was an EDM mismatch, that Fabian tried to work around.
It seems like something useful to have, but needs some work.

### Please someone work on it, because it is cool. Maybe @andiwand ?

**Comment by andiwand** [comment]
I fear I won't have time to work on this.

**Comment by FabianKlimpel** [comment]
I think I just used it for the parametrisation of hadronic interactions in Fatras. So just for a limited scope and period of time. Hence I can imangine that something went out of sync with Acts. If you want to continue using it, I recommend to ping @andriish as he is one of the HepMC developers and helped me out during the development.

**Comment by andriish** [comment]
Hi, the barcode should be used carefully or should not be used at all. Concerning other things i will be happy to help.

**Comment by andiwand** [comment]
I guess this is now fixed with your recent work on colliderml @paulgessinger ?

**Comment by paulgessinger** [comment]
Yeah I think the HepMC3 IO does what it should to the degree that it matters to us now.
